tools/monorepo-utils/src/pull-requests/commands/add-changelog-file.ts

```diff
@@ -0,0 +1,281 @@
+		Logger.warn(
+			`File 'plugins/woocommerce/changelog/pr-${ prNumber }' already exists with this content:\n`
+		);
+		log( '-------\n' + existingFileContents + '\n-------\n' );
+
+		if ( ! options.createModifyCommit ) {
+			if ( existingFileContents === changelogFileContents.trim() ) {
+				Logger.warn(
+					'The provided file contents is identical to the existing file contents, nothing to be done.'
+				);
+			}
+			log( 'Run with -c to create a commit that modifies the file.\n' );
+			exit( 0 );
+		}
+	}
+
+	// Step 4: Request the GitHub API to create the commit
+	//         (or just simulate the request if dry run requested)
+
+	const mutationCommand = `
+mutation ($input: CreateCommitOnBranchInput!) {
+	createCommitOnBranch(input: $input) { commit { url } }
+}`;
+
+	const mutationData = {
+		input: {
+			branch: {
+				repositoryNameWithOwner: `${ prData.headRepository.owner.login }/${ prData.headRepository.name }`,
+				branchName: prData.headRef.name,
+			},
+			message: {
+				headline: fileAlreadyExists
+					? 'Modify changelog file'
+					: 'Add changelog file',
+			},
+			fileChanges: {
+				additions: [
+					{
+						path: `plugins/woocommerce/changelog/pr-${ prNumber }`,
+						contents: btoa( changelogFileContents ),
+					},
+				],
+			},
+			expectedHeadOid: prData.headRefOid,
+		},
+	};
+
+	if ( options.dryRun ) {
+		log( 'Dry run, this is what would be sent to the GitHub API:\n' );
+		log(
+			JSON.stringify(
+				{
+					query: mutationCommand.trim().replace( /[\n\t]+/g, ' ' ),
+					variables: mutationData,
+				},
+				null,
+				2
+			) + '\n'
+		);
+		log(
+			'Changelog file contents (encoded as Base64 in "contents"):\n\n-------\n' +
+				changelogFileContents.trim() +
+				'\n-------\n'
+		);
+		exit( 0 );
+	}
+
+	const mutationResult: any = await gql( mutationCommand, mutationData );
+
+	Logger.notice( 'Success!' );
+	log(
+		'Commit URL:\n' + mutationResult.createCommitOnBranch.commit.url + '\n'
+	);
+
+	exit( 0 );
+};
+
+// Command declaration
+
+const commandSummary = `Adds a changelog file to a pull request.`;
+
+const commandDescription = `
+${ commandSummary }
+Uses the GitHub REST API, no need to keep a local copy of the repository.
+GITHUB_TOKEN environment variable is required.
+
+Significances:
+
+    patch or p:  Backwards-compatible bug fixes
+    minor or m:  Added (or deprecated) functionality in a backwards-compatible manner
+    major or j:  Broke backwards compatibility in some way
+
+Types:
+
+    fix or f:         Fixes an existing bug
+    add or a:         Adds functionality
+    update or u:      Update existing functionality
+    dev or d:         Development related task
+    tweak or t:       A minor adjustment to the codebase
+    performance or p: Address performance issues
+    enhancement or e: Improve existing functionality
+
+Example:
+
+    GITHUB_TOKEN=$(cat ~/my_github_key) pnpm utils pull-request add-changelog-file 12345 j a Add compatibility with Nextor
+`;
+
+const types = [
```

Comment:
Could the types be read from composer.json files under `extra` -> `change logger`?

Reply:
Maybe, but may I ask why? If there isn't a clear benefit on doing that I think that having a hardcoded list of types (which isn't going to change often anyway) is a simpler (and thus easier to maintain) approach.